File: src/player.zig
const rl = @import("raylib");
const math = @import("math.zig");

pub const Player = struct {
    position: rl.Vector2,
    size: math.Size(i32),

    pub fn init(initialPosition: rl.Vector2, size: math.Size(i32)) Player {
        return .{
            .position = initialPosition,
            .size = size,
        };
    }

    pub fn draw(self: *const Player) void {
        const x: i32 = @as(i32, @intFromFloat(self.position.x));
        const y: i32 = @as(i32, @intFromFloat(self.position.y));
        const w = self.size.width;
        const h = self.size.height;

        rl.drawRectangle(x, y, w, h, rl.Color.red);
    }
};

File: src/player.c
#include <raylib.h>
#include <stdbool.h>

#include "player.h"

#define PLAYER_W 50
#define PLAYER_H 50

Rectangle GetPlayerRect(Player *p)
{
    return (Rectangle) {
        .x = p->x,
        .y = p->y,
        .width = PLAYER_W,
        .height = PLAYER_H,
    };
}

Player CreatePlayer(int x, int y)
{
    return (Player) {
        .x = x,
        .y = y,
    };
}

void UpdatePlayer(Player *p, float dt)
{
    p->x = p->collider->position.x;
    p->y = p->collider->position.y;

    if (IsKeyDown(KEY_LEFT)) {
        p->collider->velocity.x -= 1000 * dt;
    } else if (IsKeyDown(KEY_RIGHT)) {
        p->collider->velocity.x += 1000 * dt;
    } else {
        p->collider->velocity.x = 0;
    }
}

void DrawPlayer(Player *p)
{
    DrawRectangle(p->x, p->y, PLAYER_W, PLAYER_H, GREEN);
}
